agent/engine/docker_client.go

```diff
@@ -686,33 +643,25 @@ func (dg *dockerGoClient) stopContainer(ctx context.Context, dockerID string) Do
 	return metadata
 }
 
-func (dg *dockerGoClient) RemoveContainer(dockerID string, timeout time.Duration) error {
-	// Remove a context that times out after the 'timeout' duration
-	// This is defined by 'removeContainerTimeout'. 'timeout' makes it
-	// easier to write tests
-	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
-	defer cancel()
-
+// The timeout parameter here is only used to assemble the return error message
+// TODO: replace the timeout with a string type error message
+func (dg *dockerGoClient) RemoveContainer(ctx context.Context, dockerID string, timeout time.Duration) error {
 	// Buffered channel so in the case of timeout it takes one write, never gets
 	// read, and can still be GC'd
 	response := make(chan error, 1)
-	go func() { response <- dg.removeContainer(dockerID, ctx) }()
+	go func() { response <- dg.removeContainer(ctx, dockerID) }()
 	// Wait until we get a response or for the 'done' context channel
 	select {
 	case resp := <-response:
 		return resp
 	case <-ctx.Done():
-		err := ctx.Err()
-		// Context has either expired or canceled. If it has timed out,
-		// send back the DockerTimeoutError
-		if err == context.DeadlineExceeded {
-			return &DockerTimeoutError{removeContainerTimeout, "removing"}
-		}
-		return &CannotRemoveContainerError{err}
```

Comment:
Since we are not using this error type anymore, it can be removed.

Reply:
Giving it some thought, you can capture the error [here](https://github.com/aws/amazon-ecs-agent/blob/487e699ce931a374d5e92a2ea6ddda702a637a69/agent/engine/docker_client.go#L720) and do
```
if err != nil {
    return &CannotRemoveContainerError{err}
}
```